packages/block-library/src/cover/edit.js

```diff
@@ -97,9 +113,20 @@ class CoverEdit extends Component {
 			dimRatio,
 			focalPoint,
 			hasParallax,
-			id,
 			url,
 		} = attributes;
+
+		const onSelectUrl = ( newURL ) => {
+			if ( newURL !== url ) {
+				this.props.setAttributes( {
+					url: newURL,
+					id: undefined,
+				} );
+			}
+
+			this.toggleIsEditing();
```

Comment:
I feel like it might be safer if this could be called with an argument to force the value of `isEditing` - `this.toggleIsEditing( false )`. That would specifically turn off editing mode, just in case there's a scenario where `onSelectUrl` is called when `isEditing` is `false` and results in a transition back to the placeholder.